File: src/main/java/aurocosh/divinefavor/common/lib/SuccessCounter.java
package aurocosh.divinefavor.common.lib;

public class SuccessCounter {
    private int successesNeeded;
    private int attemptsMax;

    private int successes;
    private int attempts;

    public SuccessCounter(int successesNeeded, int attemptsMax) {
        this.successesNeeded = successesNeeded;
        this.attemptsMax = attemptsMax;
        reset();
    }

    public void reset() {
        successes = 0;
        attempts = 0;
    }

    public boolean attemptOnceMore() {
        attempts++;
        if(successes >= successesNeeded)
            return false;
        return attempts < attemptsMax;
    }

    public void registerSuccess(){
        successes++;
    }

    public void registerSuccess(boolean success){
        if(success)
            successes++;
    }
}

File: src/main/kotlin/aurocosh/divinefavor/common/lib/SuccessCounter.kt
package aurocosh.divinefavor.common.lib

class SuccessCounter(private val successesNeeded: Int, private val attemptsMax: Int) {

    private var successes: Int = 0
    private var attempts: Int = 0

    init {
        reset()
    }

    fun reset() {
        successes = 0
        attempts = 0
    }

    fun attemptOnceMore(): Boolean {
        attempts++
        return if (successes >= successesNeeded) false else attempts < attemptsMax
    }

    fun registerSuccess() {
        successes++
    }

    fun registerSuccess(success: Boolean) {
        if (success)
            successes++
    }
}
